### PR TITLE
Minor changes / fixes to release 0.31.0

### PR DESCRIPTION
**Pull request by Liamsi**
See commit history

### CODE REVIEW

File: UPGRADING.md [hunk 1]
```diff
@@ -3,6 +3,24 @@
 This guide provides steps to be followed when you upgrade your applications to
 a newer version of Tendermint Core.
 
+## v0.31.0
+
+Since the pubsub no longer blocks on sending, some WS clients might stop working as expected. 
+If your WS client is not consuming events fast enough, Tendermint can terminate the subscription. 
+In this case, the WS client will receive an error with description:
```
Comment (Liamsi) [inline UPGRADING.md]: ref: https://github.com/tendermint/tendermint/pull/3417#issuecomment-472285393

There is not much people can do about this. It is merely to inform people, right? @melekes

File: CHANGELOG.md [hunk 2]
```diff
@@ -14,10 +14,14 @@ Special thanks to external contributors on this release:
 
 * Apps
 
+- [genesis] [\#2920](https://github.com/tendermint/tendermint/issues/2920) Rename `consensus_params.block_size` to `consensus_params.block` in ABCI ConsensusParams
+
 * Go API
 - [libs/common] TrapSignal accepts logger as a first parameter and does not block anymore
   * previously it was dumping "captured ..." msg to os.Stdout
   * TrapSignal should not be responsible for blocking thread of execution
+- [privval] [\#3351](https://github.com/tendermint/tendermint/pull/3351) First part of larger refactoring that clarifies and separates concerns in the privval package.
```
Comment (melekes) [inline CHANGELOG.md]: We can move it to improvements below. Even though it's breaking technically, we do not consider it "breaking" here

From readme
That said, breaking changes in the following packages will be documented in the
CHANGELOG even if they don't lead to MINOR version bumps:

- crypto
- types
- rpc/client
- config
- node
- libs
  - bech32
  - common
  - db
  - errors
  - log

Comment (Liamsi) [reply CHANGELOG.md]: Thanks! Done in https://github.com/tendermint/tendermint/pull/3422/commits/fc08a234ff78434ffca346777f04d59be14bd970

File: CHANGELOG.md [hunk 3]
```diff
@@ -49,6 +52,8 @@ Special thanks to external contributors on this release:
 - [libs/pubsub] [\#951](https://github.com/tendermint/tendermint/issues/951), [\#1880](https://github.com/tendermint/tendermint/issues/1880) use non-blocking send when dispatching messages [ADR-33](https://github.com/tendermint/tendermint/blob/develop/docs/architecture/adr-033-pubsub.md)
 - [p2p] [\#3369](https://github.com/tendermint/tendermint/issues/3369) do not panic when filter times out
 - [cmd] [\#3408](https://github.com/tendermint/tendermint/issues/3408) Fix `testnet` command's panic when creating non-validator configs (using `--n` flag) (@srmo)
+- [blockchain] [\#3358](https://github.com/tendermint/tendermint/pull/3358) Fix timer leak in `BlockPool` (@guagualvcha)
+- [p2p] [\#3359](https://github.com/tendermint/tendermint/pull/3359) Fix order of events: start peer, then add it to `PeerSet`, and then start reactors on peer. This prevents receiving to error before actually starting or adding the peer to the set. (@guagualvcha)
```
Comment (melekes) [inline CHANGELOG.md]: Why not simply "Fix reconnecting report duplicate ID error due to race condition between adding peer to peerSet and starting it"?

File: UPGRADING.md (hunk 1 above)
Comment (melekes) [reply UPGRADING.md]: Right. Normally they should not see that error.

File: CHANGELOG.md (hunk 3 above)
Comment (Liamsi) [reply CHANGELOG.md]: Thought it might be unclear what it means. But yeah, what I wrote is probably to verbose for a changelog entry.

**Pull request merged**